### DISCUSSION TITLE
Upgrade to 3.7.x error CharFields must define a 'max_length' attribute

### DISCUSSION THREAD

**Discussion by gitwpia** [netbox-community/netbox#15377] [Help Wanted!]
Upgrade to 3.7.x error CharFields must define a 'max_length' attribute

Hello
System Debian 11.9 with all update
Python 3.9.2

I always update my netbox with this method:
cd /opt/netbox
git checkout master
git pull origin master

Now I have version 3.6.6 and when I update to 3.6.9 is ok
git pull origin v3.6.9

But when I update to version 3.7.x I have error:
/var/log/syslog
```
Started netbox.
Performing system checks...
Exception in thread django-main-thread:
Traceback (most recent call last):
File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
 self.run()
  File "/usr/lib/python3.9/threading.py", line 892, in run
   self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/dist-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/django/core/management/commands/runserver.py", line 134, in inner_run
   self.check(display_num_errors=True)
  File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 546, in check
   raise SystemCheckError(msg)
 django.core.management.base.SystemCheckError: SystemCheckError: System check identified some issues:
 ERRORS:
 vpn.IKEPolicy.mode: (fields.E120) CharFields must define a 'max_length' attribute.
 vpn.IKEProposal.authentication_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
 vpn.IKEProposal.authentication_method: (fields.E120) CharFields must define a 'max_length' attribute.
 vpn.IKEProposal.encryption_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
 vpn.IPSecProfile.mode: (fields.E120) CharFields must define a 'max_length' attribute.
 vpn.IPSecProposal.authentication_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
 vpn.IPSecProposal.encryption_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
 System check identified 7 issues (0 silenced).
```

And no netbox page.

When I back - netbox work fine in v3.6.9
git reset --hard master@{"10 minutes ago"}

What to do to work fine version 3.7.x?
Thank you

**Comment by markkuleinio** [comment]
> I always update my netbox with this method:
cd /opt/netbox
git checkout master
git pull origin master

Did you follow the upgrade process? https://docs.netbox.dev/en/stable/installation/upgrading/#4-run-the-upgrade-script

**Reply by gitwpia** [reply]
Hello
I usually don't use ./upgrade.sh but this time is no error. Netbox has the same error. 
When I rollback 3.6.9 then work fine. 

Maybe version Python 3.9.2 is to low?
How to set "CharFields must define a 'max_length' attribute"
Thank you

**Reply by gitwpia** [reply]
I # line 134 (self.check(display_num_errors=True)) of file
/usr/local/lib/python3.9/dist-packages/django/core/management/commands/runserver.py

and now it's work fine...
but it's temporary solution

Is better solution?

**Comment by gitwpia** [comment]
Another problem:
```
You have 29 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): c>
Run 'python manage.py migrate' to apply them.
```

python3 manage.py migrate

```
SystemCheckError: System check identified some issues:

ERRORS:
vpn.IKEPolicy.mode: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IKEProposal.authentication_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IKEProposal.authentication_method: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IKEProposal.encryption_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IPSecProfile.mode: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IPSecProposal.authentication_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IPSecProposal.encryption_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
```

I get back to 3.6.9

**Reply by markkuleinio** [reply]
So, did you run upgrade.sh? (you didn't say)

**Reply by gitwpia** [reply]
yes with no error

**Comment by gitwpia** [comment]
My "solution"

line 134 # of file
/usr/local/lib/python3.9/dist-packages/django/core/management/commands/runserver.py

line 546 # of file
/usr/local/lib/python3.9/dist-packages/django/core/management/base.py

restart service
./upgrade

line 546 # remove
restart service
and it's work fine with no error

**Comment by gitwpia** [comment]
I try another procedure update from 3.6.9 to 3.7.4 and now is OK.
